src/profile/v100/proto_profile_maybe_with_legacy_p2p_links.rs

```diff
@@ -0,0 +1,124 @@
+use crate::prelude::*;
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoProfileMaybeWithLegacyP2PLinks {
+    #[serde(rename = "appPreferences")]
+    pub app_preferences: ProtoAppPreferencesMaybeWithLegacyP2PLinks,
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoAppPreferencesMaybeWithLegacyP2PLinks {
+    #[serde(rename = "p2pLinks")]
+    pub p2p_links: Vec<ProtoDummyLinkMaybeWithLegacyP2PLinks>,
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct ProtoDummyLinkMaybeWithLegacyP2PLinks {
+    #[serde(rename = "displayName")]
+    pub display_name: String,
```

Comment:
ah, if you actually make use of `DisplayName` type here, you can use `DisplayName::sample()` in your:

```
HasSampleValues for ProtoDummyLinkMaybeWithLegacyP2PLinks
```
which will be shorter :)